### ISSUE TITLE
namp, a cli app, build based on GUI build components!

### ISSUE THREAD

**Issue by calarndt** [d99kris/namp#2]
qmake for QT requires a huge number of dependencies all related toward the GUI (X) environment.
I use the ubuntu server dist and those dependencies really don't belong in a production server environment. My hope is to use namp for a MOH system (music on hold). However I can't get it compiled since the build system requires qmake.

**Comment by d99kris** [comment]
You can check out Namp v1, it has similar UI and functionality but was implemented in C (and no Qt dependency): https://github.com/d99kris/namp/tree/v1

**Comment by calarndt** [comment]
Thanks, Will give that a try!

**Comment by mckaygerhard** [comment]
are the v1 branch getting support? are the v1 branch with stable and receive commits? due suppost if you recomended that for, means you also support it right?

i want to give a try too ... in that case!

**Comment by calarndt** [comment]
V1 compiled fine on my server. Now I just have to go back through and remove all the stuff I added to enable building it. I runs fine and acts as I expected it too. My use case is kind of odd. I am using it to provide music for the music on hold portion of my phone system. So far it has worked flawlessly.

**Comment by d99kris** [comment]
Thanks for the update. Glad to hear it works for you @calarndt

**Comment by d99kris** [comment]
Hi @mckaygerhard - the v1 branch is not actively developed, and I don't have any current plans of supporting it to be honest. But if there's an interest from more people I might consider putting it in a separate repository and do some minimal maintenance. I initially wrote namp for my own usage, and rewrote it in C++/Qt when I got a Mac and needed it to work on both Mac and Linux. The v1 version has some library dependencies that were not easily available on Mac at the time.

**Comment by mckaygerhard** [comment]
i will package and use v1 extensively and promote it, i just fine with support bugs only.. no cares about develop new features .. just bugs solving and fine! but.. important notes here:

but @d99kris Well I thinking that what you say, i think this issue must be opened due you have a wron not so good decision .. 

lest see: qt5 consumes a lot and the level of dependencies is too a lot .. think about it! 
*a very heavy framework (yes qt5 are a king ot framework for C) with gui for an app that not have a gui!  
* you only note "mac as wonderful because qt5 you put the img and now!" but that's only up to the moment!


* glib2.0: I imagine that to make the C code easier ... that exists in mac ...
* ncursesw5: if the interface I imagine this is the most lol but available for mac
* mpg123, libao, alsa, tag openssl libssl: here the problem, all can be changed to LAME!

**so with more work, just Lame, glib2 and ncurses. those 3 are available for mac easyle!.**. but well qt5 ? recode all in C++ and latr will code again ...
so..  when qt5 will deprecated and comes qt6 will stops supporting your mac .. must change the os due "are not supported anymore" etc etc so will code again? and again? and again?

I do not like that behavior changes and changes as if it were a fashion .. now about the dependencies in mac, it is simple includes the code of the libs in the project and that are compiled statically .. It is also a question of analyzing more! so you dont make a good decition on qt5.

**Comment by d99kris** [comment]
I think you nailed my main concern with "more work" in "so with more work, just Lame, glib2 and ncurses [it can be done]". I have a job and also other github projects, i.e. limited time.

You mentioned you are planning to package namp. May I ask for which platform / package manager?

Anyway, I reported a new issue https://github.com/d99kris/namp/issues/3 to gauge the interest of maintaining a non-Qt version of namp (this will be in parallel to main namp which will still be Qt-based, and only for Linux platform). Please "like" that issue, and ask anyone who wants this too to like it. If there's sufficient interest I'll start maintaining v1 (for Linux only) in a separate repository.

**Comment by mckaygerhard** [comment]
1.- umm i cannot understand very well the meaning of your english las response about "nailed".. measn "pickup a point" or means "you are confused about it" ?

2.- i`ll package for debian but debian 6 squeeze very oldstable.. due "if are working dont change it" and debian 6 its the maximun debian verison that works in my zohan cam grabber and my other capture cards...

3.- great! ..

**Comment by d99kris** [comment]
1. Yeah I meant in the sense of "pickup a point".

2. Ok cool.

The repository https://github.com/d99kris/namp-lite has now been created for maintaining a non-Qt version of namp. Thanks for all input.
